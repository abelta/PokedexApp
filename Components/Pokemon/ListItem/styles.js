import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  listItem: {
    borderColor: 'lightgrey',
    borderWidth: StyleSheet.hairlineWidth,
    height: 35,
    justifyContent: 'center',
  },
  text: {
    fontSize: 18,
    textAlign: 'center',
  },
});

export default styles;
