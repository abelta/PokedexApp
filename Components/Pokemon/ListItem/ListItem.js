import React from 'react';
import { Text, View } from 'react-native';
import { Link } from 'react-router-native';
import { pokemonEntryType } from '../../../PropTypes/Pokemon';
import styles from './styles';

const ListItem = ({ pokemonEntry }) => (
  <View style={styles.listItem}>
    <Link to={`/pokemon/${pokemonEntry.name}`}>
      <Text style={styles.text}>{pokemonEntry.name}</Text>
    </Link>
  </View>
);

ListItem.propTypes = {
  pokemonEntry: pokemonEntryType.isRequired,
};

export default ListItem;
