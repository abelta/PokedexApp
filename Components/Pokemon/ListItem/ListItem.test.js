import React from 'react';
import { MemoryRouter } from 'react-router-native';
import renderer from 'react-test-renderer';

import ListItem from '.';

describe('ListItem', () => {
  it('renders properly', () => {
    const pokemonEntry = { name: 'pikachu' };
    const tree = renderer.create(
      <MemoryRouter>
        <ListItem pokemonEntry={pokemonEntry} />
      </MemoryRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
