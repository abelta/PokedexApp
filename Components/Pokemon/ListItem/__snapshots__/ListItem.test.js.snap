// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem renders properly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderColor": "lightgrey",
      "borderWidth": 0.5,
      "height": 35,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 18,
        "textAlign": "center",
      }
    }
  >
    pikachu
  </Text>
</View>
`;
